src/topology/algebra/uniform_extension.lean

```diff
@@ -0,0 +1,253 @@
+/-
+Copyright (c) 2019 Zhouhang Zhou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Zhouhang Zhou
+-/
+
+import topology.uniform_space.uniform_embedding
+import topology.algebra.module
+import analysis.normed_space.operator_norm
+
+/-!
+# Extension of uniform continuous functions
+
+Extension of a uniform continuous function `f : β → γ` on a uniform space `β` to its completion
+preserves many of `f`'s properties.
+
+## Main statements
+
+Let `α` and `β` be uniform spaces and let `β` be a dense subset of `α`, with `e : β → α` as the
+inclusion map. Let `f : β → γ` be a uniform continuous function from `β` to a complete Hausdorff
+space `γ`. Then there is a uniform continuous function `ψ : α → γ` on `α`, such that `ψ` is an
+extension of `f`, i.e. `ψ ∘ e = f`.
+
+In this file we show that
+
+* If `f` is an `add_monoid_hom`, then `ψ` is also an `add_monoid_hom`.
+* If `f` is a `continuous_linear_map`, then `ψ` is also a `continuous_linear_map`.
+* If `∥f∥ ≤ M`, then `∥ψ∥ ≤ M`.
+
+## Main definitions
+
+* For `f : β →+ γ`, `f.extend _ _ _ _ : α →+ γ` is the extension of `f`.
+* Fof `f : α →L[K] γ`, `f.extend _ _ _ _ : α →L[K] γ` is the extension of `f`.
+
+## Implementation notes
+
+Mainly used in `measure_theory.bochner_integration` to extend integrals on integrable simple
+functions to integrals on all integrable functions.
+
+One can find similar results in `topology.algebra.group_completion` and
+`topology.algebra.uniform_ring`, where an explicit construction of completions is given, and thus
+`f : β → γ` is extended to `ψ : (completion β) → γ`. This approach is not suitable for extending
+integrals, since integrable functions are usually defined independently of simple functions.
+
+For results about functions without extra structures attached to them, see
+`topology.uniform_space.abstffract_completion`.
```

Comment:
```suggestion
`topology.uniform_space.abstract_completion`.
```

Reply:
I don't understand why things went wrong. This is the only change and it is in comments.

Reply:
In this case, I suspect [your build timed out](https://travis-ci.org/leanprover-community/mathlib/jobs/609968597) because mathlib was updated between stages, so it had to completely rebuild from scratch. See the (long) discussion [in this Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Travis/near/178966978) from several weeks ago. When this happens, you can ping someone to restart the build. 

Probably you should be added as a "collaborator" to mathlib, which will give you the power to restart / cancel builds on Travis yourself... I'm not sure who's in charge of that now, maybe @robertylewis ?

Reply:
@aceg00 you have a collaborator invitation now.

Reply:
Oh, this is great! Thank you.